pkg/gds/admin_test.go

```diff
@@ -0,0 +1,939 @@
+package gds
```

Comment:
I didn't want to put this in the same package as gds but I couldn't figure out a good way to replace the underlying database with the test fixture one because the `db` field is not exported. I'm definitely open to other solutions on how to deal with this.